x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/components/tag_options.tsx

```diff
@@ -78,29 +108,50 @@ export const TagOptions: React.FC<Props> = ({ tagName, isTagHovered, onTagsUpdat
         >
           <EuiFlexGroup direction="column" alignItems="flexStart" gutterSize="xs">
             <EuiFlexItem>
-              <EuiFieldText
-                placeholder={i18n.translate('xpack.fleet.tagOptions.nameTextFieldPlaceholder', {
-                  defaultMessage: 'Enter new name for tag',
-                })}
-                value={updatedName}
-                required
-                onChange={(e: any) => {
-                  const newName = e.target.value;
-                  setUpdatedName(newName);
-                  if (!newName) {
-                    return;
-                  }
-                  debouncedSendRenameTag(newName);
-                }}
-              />
+              <EuiFlexGroup gutterSize="xs">
+                <EuiFlexItem>
+                  <EuiFieldText
+                    placeholder={i18n.translate('xpack.fleet.tagOptions.nameTextFieldPlaceholder', {
+                      defaultMessage: 'Enter new name for tag',
+                    })}
+                    value={updatedName}
+                    required
+                    onKeyDown={(e: { key: string }) => {
+                      if (e.key === 'Enter') {
+                        handleRename(updatedName!);
+                        closePopover();
+                      }
+                    }}
+                    onChange={(e: any) => {
```

Comment:
Do we need the `: any` and type assertion here + above? I would expect these EUI components to expose type definitions for these events.